### PR TITLE
Adding support for explicit type signatures

### PR DESCRIPTION
**Pull request by viktorcsimma**
This is accomplished by the `the` function in Haskell.Prim. The first argument is the type, the second one is the value itself. For example, `the Nat 5` gets compiled to `5 :: Natural`.

Let me know what you think.

### CODE REVIEW

File: src/Agda2Hs/Compile/Term.hs [hunk 1]
```diff
@@ -68,7 +73,18 @@ ifThenElse _ es = compileElims es >>= \case
...
   -- partially applied
-  _ -> genericError $ "if_then_else must be fully applied"
+  _ -> genericError $ "if_then_else_ must be fully applied"
+
+expTypeSig :: QName -> Elims -> C (Hs.Exp ())
+expTypeSig _ es = do
+  let args = fromMaybe __IMPOSSIBLE__ $ allApplyElims es
+  case args of
+    _ : typArg : expArg : args' -> do         -- the first one is the level; we throw that away
+      exp <- fromMaybe __IMPOSSIBLE__ <$> compileArg expArg -- this throws an error if it was Nothing
```
Comment (jespercockx) [inline src/Agda2Hs/Compile/Term.hs:83]: You're better off just doing `compileTerm . unArg`, which would avoid the call to `__IMPOSSIBLE__`

File: test/The.agda [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module The where
```
Comment (jespercockx) [inline test/The.agda:1]: Maybe a name such as `TypeSignature` for the module would be more helpful?

**Review by jespercockx — APPROVED** [review]
Thanks for the PR! I have two small remarks but otherwise this seems like a really good feature to have. Since I've merged some other PRs in the meantime, could you also please merge/rebase on the current master?

**Comment by viktorcsimma** [comment]
Done:)

**Review by jespercockx — APPROVED** [review]
Thank you!

**Pull request merged**